python_modules/dagster/dagster/_cli/proxy_server_manager.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import os
+import threading
+from collections.abc import Mapping, Sequence
+from contextlib import AbstractContextManager, ExitStack
+from typing import Any, Union, cast
+
+from typing_extensions import Self
+
+import dagster._check as check
+from dagster._core.errors import DagsterUserCodeUnreachableError
+from dagster._core.instance import DagsterInstance
+from dagster._core.remote_representation.grpc_server_registry import GrpcServerRegistry
+from dagster._core.remote_representation.origin import (
+    GrpcServerCodeLocationOrigin,
+    ManagedGrpcPythonEnvCodeLocationOrigin,
+)
+from dagster._core.workspace.context import WEBSERVER_GRPC_SERVER_HEARTBEAT_TTL
+from dagster._core.workspace.load_target import WorkspaceLoadTarget
+from dagster._grpc.server import INCREASE_TIMEOUT_DAGSTER_YAML_MSG, GrpcServerCommand
+
+
+def get_auto_restart_code_server_interval() -> int:
+    return int(os.getenv("DAGSTER_CODE_SERVER_AUTO_RESTART_INTERVAL", "30"))
+
+
+class ProxyServerManager(AbstractContextManager):
+    """Context manager that manages the lifecycle of a set of proxy code servers targeting a passed-in load target."""
+
+    def __init__(
+        self,
+        instance: DagsterInstance,
+        workspace_load_target: WorkspaceLoadTarget,
+        code_server_log_level: str = "INFO",
+        server_command: GrpcServerCommand = GrpcServerCommand.API_GRPC,
```

Comment:
do we need this? isn't it always going to be the same thing?

Reply:
Yea I thought i already got rid of this tbh. And the name gives it away anyway